src/startfile/utils.py

```diff
@@ -15,23 +15,25 @@ def startfile(value: Union[str, Path]):
 
     path = value
 
-    if platform.system() == "Darwin":
-        return startfile_macos(path)
+    return startfile_posix(path) if os.name == "posix" else startfile_windows(path)
 
-    if platform.system() == "Windows":
-        return startfile_windows(path)
 
-    return startfile_linux(path)
+def startfile_posix(path):
+    command = "open" if platform.system() == "Darwin" else "xdg-open"
```

Comment:
I would prefer these remain separate functions. macOS is POSIX-compliant but it's still a different command.